modules/machineset-yaml-osp-sr-iov.adoc

```diff
@@ -0,0 +1,111 @@
+    machine.openshift.io/cluster-api-machine-role: <node_role>
+    machine.openshift.io/cluster-api-machine-type: <node_role>
+  name: <infrastructure_ID>-<node_role>
+  namespace: openshift-machine-api
+spec:
+  replicas: <number_of_replicas>
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+      machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>
+  template:
+    metadata:
+      labels:
+        machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+        machine.openshift.io/cluster-api-machine-role: <node_role>
+        machine.openshift.io/cluster-api-machine-type: <node_role>
+        machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>
+    spec:
+      metadata:
+      providerSpec:
+        value:
+          apiVersion: openstackproviderconfig.openshift.io/v1alpha1
+          cloudName: openstack
+          cloudsSecret:
+            name: openstack-cloud-credentials
+            namespace: openshift-machine-api
+          flavor: <nova_flavor>
+          image: <glance_image_name_or_location>
+          serverGroupID: <optional_UUID_of_server_group>
+          kind: OpenstackProviderSpec
+          networks:
+            - subnets:
+              - UUID: <machines_subnet_UUID>
+          ports:
+            - networkID: <radio_network_UUID> <1>
+              nameSuffix: radio
+              fixedIPs:
+                - subnetID: <radio_subnet_UUID> <2>
+              tags:
+                - sriov
+                - radio
+              vnicType: direct <3>
+              portSecurity: false <4>
+            - networkID: <uplink_network_UUID> <1>
+              nameSuffix: uplink
+              fixedIPs:
+                - subnetID: <uplink_subnet_UUID> <2>
+              tags:
+                - sriov
+                - uplink
+              vnicType: direct <3>
+              portSecurity: false <4>
+          primarySubnet: <machines_subnet_UUID>
+          securityGroups:
+          - filter: {}
+            name: <infrastructure_ID>-<node_role>
+          serverMetadata:
+            Name: <infrastructure_ID>-<node_role>
+            openshiftClusterID: <infrastructure_ID>
+          tags:
+          - openshiftClusterID=<infrastructure_ID>
+          trunk: true
+          userDataSecret:
+            name: <node_role>-user-data
+          availabilityZone: <optional_openstack_availability_zone>
+          configDrive: true <5>
+----
+<1> Enter a network UUID for each port.
+<2> Enter a subnet UUID for each port.
+<3> The value of the `vnicType` parameter must be `direct` for each port.
+<4> The value of the `portSecurity` parameter must be `false` for each port.
++
+Security groups and allowed address pairs cannot be set for ports where port security is disabled. Setting security groups on the instance applies the groups to all ports that are attached to it.
```

Comment:
I am not familiar with the topic, so I'm wrestling a little with "where port security is disabled" because my brain wants to read "when".  I also would have loved to read "You cannot..." but again, I do not know the topic.

Reply:
```suggestion
You cannot set security groups and allowed address pairs for ports when port security is disabled. Setting security groups on the instance applies the groups to all ports that are attached to it.
```